### ISSUE TITLE
RFE: set RSTRNT_TASKNAME based on TMT_TEST_NAME

### ISSUE THREAD

**Issue by bgoncalv** [teemtee/tmt#3041]
A lot of legacy tests expect `RSTRNT_TASKNAME` to be set.

It would be nice if tmt could set this variable. I think it could match `TMT_TEST_NAME`.

Looking at https://tmt.readthedocs.io/en/stable/overview.html#test-variables it seems similar approach is already done with `RSTRNT_REBOOTCOUNT`

**Comment by happz** [comment]
This one should be fairly easy to achieve. Adding to the triage meeting though, I'm not sure whether we want to hide it behind the `restraint-compatible` flag - I would hide it, yet `RSTRNT_REBOOTCOUNT` isn't...

**Comment by psss** [comment]
Yes, all newly added backward-compatibility features should be implemented under the `restraint-compatible` flag. Some old ones did not make it there, we should probably move them all under it. But that would be a backward-incompatble change, so it should be handle with care:

* #4021
